include/seastar/testing/perf_tests.hh

```diff
@@ -167,23 +167,26 @@ class time_measurement {
     perf_stats _start_stats;
     perf_stats _total_stats;
 
+    linux_perf_event* _instructions_retired_counter;
+
 public:
     [[gnu::always_inline]] [[gnu::hot]]
     void start_run(linux_perf_event* instructions_retired_counter = nullptr) {
+        _instructions_retired_counter = instructions_retired_counter;
         _total_time = { };
         auto t = clock_type::now();
         _run_start_time = t;
         _start_time = t;
-        _start_stats = perf_stats::snapshot(instructions_retired_counter);
+        _start_stats = perf_stats::snapshot(_instructions_retired_counter);
     }
 
     [[gnu::always_inline]] [[gnu::hot]]
-    performance_test::run_result stop_run(linux_perf_event* instructions_retired_counter = nullptr) {
+    performance_test::run_result stop_run() {
```

Comment:
It is not clear to me why the existing code always takes this `linux_perf_event` pointer for all of the methods, but I couldn't find any way to pass this from within a test. It seems safe (lifetime-wise) and more ergonomic to just capture it in `time_measurement`.

Reply:
IIRC the reason was that it was introduced as an add-on feature on top of the original logic that didn't measure cpu instructions, and being reluctant to introduce state to `time_measurement`.

Reply:
Probably it should have been a constructor parameter to time_measurement.